app/controllers/work_packages/bulk_controller.rb

```diff
@@ -39,34 +39,18 @@ class WorkPackages::BulkController < ApplicationController
   include IssuesHelper
 
   def edit
-    @work_packages.sort!
-    @available_statuses = @projects.map { |p|Workflow.available_statuses(p) }.inject { |memo, w|memo & w }
-    @custom_fields = @projects.map(&:all_work_package_custom_fields).inject { |memo, c|memo & c }
-    @assignables = @projects.map(&:possible_assignees).inject { |memo, a| memo & a }
-    @responsibles = @projects.map(&:possible_responsibles).inject { |memo, a| memo & a }
-    @types = @projects.map(&:types).inject { |memo, t| memo & t }
+    WorkPackageBulkUpdateService.new(@work_packages, @projects).run(params)
+      .available_attributes.each do |key, value|
+        instance_variable_set(:"@#{key}", value)
+      end
```

Comment:
I don't see why calling `instance_variable_set` in a loop is a better idea than explicitly setting three instance variables by hand.